packages/instant-meilisearch/src/adapter/search-request-adapter/filter-adapter.ts

```diff
@@ -1,93 +1,155 @@
+  // Typescript has problems with readonly, which makes it undesirable in
+  // most situations, the crux of the issue being that `Array.isArray`
+  // will return true for mutable arrays as well as immutable ones. Since Instantsearch.js
+  // decided to use readonly, we have to type cast, while keeping as much of the type
+  // safety as possible.
+  return typeof filters === 'string'
+    ? transformCallback(filters)
+    : filters
+        .map((filter) =>
+          (<(value: readonly any[] | any) => value is readonly any[]>(
+            Array.isArray
+          ))(filter)
+            ? filter
+                .map((nestedFilter) => transformCallback(nestedFilter))
+                // TODO: Do these filters have any purpose? Can we actually get empty strings? Should we handle empty strings?
+                //       Maybe handling these should be the responsibility of instantsearch.js and/or the user and their input.
+                //       Instead of these empty filters quietly being swallowed, they might even get us an error message, and
+                //       we might know something is wrong?
+                .filter((elem) => elem !== '')
+            : transformCallback(filter)
+        )
+        // works on strings too, as they're somewhat of an array
+        .filter((elem) => elem.length !== 0)
 }
 
 /**
  * Return the filter in an array if it is a string
  * If filter is array, return without change.
  *
- * @param  {Filter} filter
- * @returns {Array}
+ * @param  {Filter} [filter]
+ * @returns {Array|undefined}
  */
-function filterToArray(filter: Filter): Array<string | string[]> {
-  // Filter is a string
-
-  if (filter === '') return []
-  else if (typeof filter === 'string') return [filter]
-  // Filter is either an array of strings, or an array of array of strings
-  return filter
+function nonEmptyFilterToArray(
+  filter?: Filter
+): Array<string | string[]> | undefined {
+  return filter !== undefined && filter.length !== 0
+    ? // Filter is a string
+      typeof filter === 'string'
+      ? [filter]
+      : // Filter is either a one- or two-dimensional array of strings
+        filter
+    : undefined
 }
 
 /**
- * Merge facetFilters, numericFilters and filters together.
+ * Merge filters, transformedNumericFilters and transformedFacetFilters
+ * together.
  *
- * @param  {Filter} facetFilters
- * @param  {Filter} numericFilters
  * @param  {string} filters
+ * @param  {Filter} transformedNumericFilters
+ * @param  {Filter} transformedFacetFilters
  * @returns {Filter}
  */
 function mergeFilters(
-  facetFilters: Filter,
-  numericFilters: Filter,
-  filters: string
+  filters?: string,
+  transformedNumericFilters?: Filter,
+  transformedFacetFilters?: Filter
 ): Filter {
-  const adaptedFilters = filters.trim()
-  const adaptedFacetFilters = filterToArray(facetFilters)
-  const adaptedNumericFilters = filterToArray(numericFilters)
+  // TODO: If we are trimming this, shouldn't we trim the other ones too?
```

Comment:
I guess we should, yes.

Reply:
I'll look into it more, because I am unsure if trimming is needed at all.

Reply:
Trimming is unnecessary, IS.js should not send trailing spaces, and if it does I believe it's better to leave it as is. Throughout using this patched version, filters was always undefined, and the rest of the filters never had any trailing white space.